docs/source/architecture/slashing.rst

```diff
@@ -19,16 +21,30 @@ TBD (https://github.com/nucypher/nucypher/issues/803)
 How slashing affects stake
 --------------------------
 
-The goal of slashing is to reduce the number of tokens that belongs to a staking offender.
-In this case, the main task is not to violate the logic of locking tokens.
-The entire stake consists of:
+An entire stake consists of:
 
     * unlocked tokens which the staker can withdraw at any moment
     * tokens locked for a specific period
 
-.. _`Sub-stakes`: https://docs.nucypher.com/en/latest/architecture/sub_stakes.html
+In terms of how the stake is slashed, unlocked tokens are the first portion of the stake to be slashed. After that, if necessary, locked sub-stakes are decreased in order based on their remaining lock time, beginning with the shortest. The shortest sub-stake is decreased, and if the adjustment of that sub-stake is insufficient to fulfil the required punishment sum, then the next shortest sub-stake is decreased, and so on. The sub-stakes for each period are checked separately.
+
+Overall the slashing algorithm is as follows:
+
+#. Reduce unlocked tokens
+
+#. If insufficient, slash sub-stakes as follows:
+
+    a. Calculate the maximum allowed total stake for any period for the staker ::
+
+        max_allowed_stake = pre_slashed_total_stake - slashing_amount
+
+       Therefore, for any period moving forward the sum of sub-stakes for that period cannot be more than ``max_allowed_stake``.
+    b. For each period moving forward ensure that the amount of locked tokens is less than or equal to ``max_allowed_stake``. If not, then reduce the shortest sub-stake to ensure that this occurs; then the next shortest and so on, as necessary for the period.
+    c. Since sub-stakes can extend over multiple periods and can only have a single fixed amount of tokens for all applicable periods (see `Sub-stakes`_), the resulting amount of tokens remaining in a sub-stake after slashing is the minimum amount of tokens it can have across all of its relevant periods. To clarify, suppose that a sub-stake is locked for periods ``n`` and ``n+1``, and the slashing algorithm first determines that the sub-stake can have 10 tokens in period ``n``, but then it can only have 5 tokens in period ``n+1``. In this case, the sub-stake will be slashed to have 5 tokens in both periods ``n`` and ``n+1``.
+    d. The above property of sub-stakes means that there is the possibility that the total amount of locked tokens for a particular period could be reduced to even lower than the ``max_allowed_stake`` when unnecessary. Therefore, the algorithm may create new sub-stakes on the staker's behalf to utilize tokens locked in the future during earlier periods, so that the staker is not excessively punished in earlier periods. This benefits both the staker, by ensuring that their remaining tokens are efficiently utilized, and the network by maximizing its health.
```

Comment:
@vzotova Question: am I right in saying that sub-stakes only get created in `earlier periods`? Is there a case where we would create new sub-stakes in future periods...I couldn't think of such a scenario.